java/system-test/system-test-suite/src/test/java/sleeper/systemtest/suite/TableOfflineIT.java

```diff
@@ -0,0 +1,125 @@
+import sleeper.core.schema.Schema;
+import sleeper.core.statestore.FileReference;
+import sleeper.core.statestore.FileReferenceFactory;
+import sleeper.systemtest.dsl.SleeperSystemTest;
+import sleeper.systemtest.dsl.extension.AfterTestPurgeQueues;
+import sleeper.systemtest.dsl.extension.AfterTestReports;
+import sleeper.systemtest.dsl.reporting.SystemTestReports;
+import sleeper.systemtest.dsl.sourcedata.RecordNumbers;
+import sleeper.systemtest.suite.fixtures.SystemTestSchema;
+import sleeper.systemtest.suite.testutil.SystemTest;
+
+import java.nio.file.Path;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.LongStream;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static sleeper.configuration.properties.instance.CdkDefinedInstanceProperty.COMPACTION_JOB_QUEUE_URL;
+import static sleeper.configuration.properties.table.TableProperty.COMPACTION_FILES_BATCH_SIZE;
+import static sleeper.configuration.properties.table.TableProperty.PARTITION_SPLIT_THRESHOLD;
+import static sleeper.configuration.properties.table.TableProperty.TABLE_ONLINE;
+import static sleeper.core.testutils.printers.FileReferencePrinter.printFiles;
+import static sleeper.core.testutils.printers.PartitionsPrinter.printPartitions;
+import static sleeper.systemtest.dsl.sourcedata.GenerateNumberedValue.addPrefix;
+import static sleeper.systemtest.dsl.sourcedata.GenerateNumberedValue.numberStringAndZeroPadTo;
+import static sleeper.systemtest.dsl.sourcedata.GenerateNumberedValueOverrides.overrideField;
+import static sleeper.systemtest.suite.fixtures.SystemTestInstance.MAIN;
+import static sleeper.systemtest.suite.testutil.PartitionsTestHelper.partitionsBuilder;
+
+@SystemTest
+public class TableOfflineIT {
+    @TempDir
+    private Path tempDir;
+    private PartitionTree expectedPartitions;
+    private FileReferenceFactory fileFactory;
+
+    @BeforeEach
+    void setUp(SleeperSystemTest sleeper, AfterTestReports reporting, AfterTestPurgeQueues purgeQueues) {
+        sleeper.connectToInstance(MAIN);
+        reporting.reportIfTestFailed(SystemTestReports.SystemTestBuilder::compactionTasksAndJobs);
+        purgeQueues.purgeIfTestFailed(COMPACTION_JOB_QUEUE_URL);
+        expectedPartitions = partitionsBuilder(sleeper).singlePartition("root").buildTree();
+        fileFactory = FileReferenceFactory.from(expectedPartitions);
+    }
+
+    @Test
+    void shouldNotSplitPartitionsIfTableIsOffline(SleeperSystemTest sleeper) {
+        // Given
+        sleeper.updateTableProperties(Map.of(
+                PARTITION_SPLIT_THRESHOLD, "20",
+                TABLE_ONLINE, "false"));
+        sleeper.setGeneratorOverrides(
+                overrideField(SystemTestSchema.ROW_KEY_FIELD_NAME,
+                        numberStringAndZeroPadTo(2).then(addPrefix("row-"))));
+        sleeper.ingest().direct(tempDir).numberedRecords(LongStream.range(0, 100));
+
+        // When
+        sleeper.partitioning().split();
```

Comment:
Looking at the driver, it has its own logic to check whether it expects splits to occur, and if it doesn't expect any it doesn't wait for them. That means if the splitter did something different than the driver expected, it would still be running when this step returns. The assertions below can still pass even if it does go on to split the partition.

I'm not sure how we can improve on this because we can't really examine the state of the queue, and that's the only place splitting jobs are tracked. If we don't track the jobs directly we're left trying to prove a negative, waiting for nothing to happen.